### PR TITLE
Simple project (KCC Solution)

### PR DESCRIPTION
**Pull request by trodge**
This is a proposed implementation of a sample kpt package for creating a project in GCP using Config Connector CRDs

### CODE REVIEW

**Comment by trodge** [comment]
@googlebot I consent.

**Comment by BrianPi** [comment]
@googlebot I consent.

File: config-connector/solutions/projects/kpt/simple-project/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  simple-project
+
+# SYNOPSIS
+
+  Config Connector compatible YAML files to create
+  a project in an organization. Make sure your Config Connector
+  service account has the project creator role for your organization.
+
+  In order to use, replace the `${BILLING_ACCOUNT_ID?}` and `${ORG_ID?}` values
```
Comment (maqiuyujoyce) [inline config-connector/solutions/projects/kpt/simple-project/README.md]: You may want to add commands to set billing account ID and org ID here to better describe the usage.

File: config-connector/solutions/projects/kpt/simple-project/project.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (maqiuyujoyce) [inline config-connector/solutions/projects/kpt/simple-project/project.yaml]: Could you add the copyright above?

**Comment by maqiuyujoyce** [comment]
@morgante They are two separate solutions under the same solution area, projects. #598 is for project hierarchy and this one is for simple project.

**Comment by AlexBulankou** [comment]
> @morgante They are two separate solutions under the same solution area, projects. #598 is for project hierarchy and this one is for simple project.

@trodge, @maqiuyujoyce , @BrianPi  If you agree, I'd say let's have project-hierarchy go first since it is further along and then this one will have to merge before submitting?

File: config-connector/solutions/projects/kpt/simple-project/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  with your billing account and organization id numbers. You can do this with kpt setters or manually. To use kpt setters:
+  
+  ```
+  kpt cfg set billing-account <insert billing account here>
```
Comment (maqiuyujoyce) [inline config-connector/solutions/projects/kpt/simple-project/README.md]: The dir param is missing in the commands.

Comment (trodge) [reply config-connector/solutions/projects/kpt/simple-project/README.md]: Yes, you are correct, I will edit that.

File: config-connector/solutions/projects/kpt/simple-project/project.yaml (hunk 2 above)
Comment (trodge) [reply config-connector/solutions/projects/kpt/simple-project/project.yaml]: Yes, added.

File: config-connector/solutions/projects/kpt/simple-project/README.md (hunk 1 above)
Comment (trodge) [reply config-connector/solutions/projects/kpt/simple-project/README.md]: Added, now fixing syntax in next issue.

File: config-connector/solutions/projects/kpt/simple-project/Kptfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          value: "${ORG_ID?}"
+          setBy: PLACEHOLDER
+    io.k8s.cli.setters.project-id:
+      description: ID of project
+      x-k8s-cli:
+        setter:
+          name: project-id
+          value: simple-project
+          setBy: package-default
+    io.k8s.cli.setters.project-name:
```
Comment (morgante) [inline config-connector/solutions/projects/kpt/simple-project/Kptfile]: Can we default this to be the same as project-id?

Comment (trodge) [reply config-connector/solutions/projects/kpt/simple-project/Kptfile]: Yes, currently they both have value of `simple-project`. Should I make them use the same setter and therefore always have the same value?

Comment (morgante) [reply config-connector/solutions/projects/kpt/simple-project/Kptfile]: The ideal would be that setting `project-id` will set both, while setting `project-name` would override the project name only.

Comment (trodge) [reply config-connector/solutions/projects/kpt/simple-project/Kptfile]: The best way I can think of to implement this behavior with kpt would be to have them both be set by the project-id setter and make a comment in project.yaml noting that the spec.name can be changed independently of the metadata.name.

Comment (morgante) [reply config-connector/solutions/projects/kpt/simple-project/Kptfile]: Could we ask kpt team about multi-setters or something like this? It seems like a reasonable use case.

Comment (trodge) [reply config-connector/solutions/projects/kpt/simple-project/Kptfile]: I have created an issue for it [here](https://github.com/GoogleContainerTools/kpt/issues/430).

Comment (trodge) [reply config-connector/solutions/projects/kpt/simple-project/Kptfile]: @morgante in the mean time I have made them both use the same setter. Please take a look when you get the chance.

File: config-connector/solutions/projects/kpt/simple-project/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+# REQUIREMENTS
+  A working cluster with Config Connector installed.
+  
+  Cloud Resource Manager API and Cloud Billing API enabled in the project where Config Connector is installed.
+  
+  The cnrm-system service account must have `roles/resourcemanager.projectCreator` in your organization and `roles/billing.user` for your billing account.
+  
+# USAGE
+  In order to use, replace the `${BILLING_ACCOUNT_ID?}` and `${ORG_ID?}` values with your billing account and organization id. You can do this with kpt setters or by manually editing the YAML files. To use kpt setters:
+  ```
```
Comment (maqiuyujoyce) [inline config-connector/solutions/projects/kpt/simple-project/README.md]: "You can do this with kpt setters or by manually editing the YAML files. To use kpt setters:" -> "You can do this with kpt setters:"

Let's not suggest manual edits if it is not required.

**Comment by trodge** [comment]
@AlexBulankou @maqiuyujoyce I believe I have addressed all comments, take a look when you get a chance.

**Pull request merged**